packages/testing/src/execution_testing/specs/benchmark.py

```
@@ -290,6 +357,22 @@ def generate_blocks_from_code_generator(self) -> List[Block]:
 
         return [execution_block]
 
+    def generate_fixed_opcode_count_transactions(self) -> List[Block]:
+        """Generate transactions with a fixed opcode count."""
+        if self.code_generator is None:
+            raise Exception("Code generator is not set")
+        self.code_generator.deploy_fix_count_contracts(
+            pre=self.pre, fork=self.fork
+        )
+        gas_limit = (
+            self.fork.transaction_gas_limit_cap() or self.gas_benchmark_value
+        )
```

We should add a TODO here for Osaka support or consideration. If the number of fixed opcodes need more than 16M gas then we will fail. It would be nice to fail early here if we reach the tx gas limit cap.

I'm not sure if splitting into multiple txs makes much sense here for the gas repricing regression use case. Adding multiple txs would skew the timings I feel. Maybe they could be split into 2 categories. Single tx and multi tx. It would be good to know if multi txs should be supported here for forks >= Osaka.

The other consideration would be if you wanted to split them into multi txs equally... or have 1/many full and with the final tx partially full.

We could potentially find a way to disable the cap for testing in clients, but this would add a security concern.

---

AFAIK, we’re planning to disable the transaction gas limit cap for the repricing experiment. Maria hope to exclude the cap during timing measurements, and Nethermind and Geth will add support for this in internal testing.